pkg/plugin/cmd/agent/upgrade/upgrade.go

```diff
@@ -0,0 +1,165 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-2020 Datadog, Inc.
+
+package upgrade
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/DataDog/datadog-operator/pkg/apis/datadoghq/v1alpha1"
+	"github.com/DataDog/datadog-operator/pkg/plugin/common"
+
+	"github.com/spf13/cobra"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var (
+	image          string
+	upgradeExample = `
+  # upgrade the version of the datadog agent to latest
+  %[1]s upgrade
+
+  # upgrade the version of the datadog agent to latest defined in DatadogAgent foo
+  %[1]s upgrade foo
```

Comment:
sorry, but I dont get it. "foo" is the name of the DatadogAgent resource? 
what are the difference with the previous command? 

Last question what is "latest defined" version? is it the latest version supported by the operator? the plugin? how it works?

Reply:
> what are the difference with the previous command?
when multiple DatadogAgent resources are applied, the first command will upgrade all of them, but if the DatadogAgent is mentioned the command upgrades only the image of the chosen DatadogAgent

> Last question what is "latest defined" version?
the wording is a bit confusing i'll update it, but by latest I mean the image tag `latest`, and the `version of the datadog agent` is defined in `DatadogAgent foo` 

although I guess we need to print a warning if we find the current deployed tag is already `latest` - and make it clear we're not going to trigger an upgrade